docs/upgrade.html

```diff
@@ -47,14 +47,77 @@
 
 <p><b>Note:</b> Bumping the protocol version and restarting can be done any time after the brokers were upgraded. It does not have to be immediately after.
 
-<ol>
+<h5><a id="upgrade_1020_streams" href="#upgrade_1020_streams">Updating a Kafka Streams Application</a></h5>
+<ul>
     <li>Upgrading a Kafka Streams Applications:
         <ul>
-            <li>You need to recompile your code. Just swapping the jar file will not work and will break your appliation.</li>
-            <li>If you use a custom timestamp extractor, you will need to update this code, because the <code>TimestampExtractor</code> interface was changed.</li>
+            <li> You need to recompile your code. Just swapping the jar file will not work and will break your application. </li>
+            <li> <code>KStreamBuilder#table()</code> always requires to specify a store name. </li>
+            <li> <code>KTable#through()</code> always requires to specify a store name. </li>
+            <li> If you use a custom timestamp extractor, you will need to update this code, because the <code>TimestampExtractor</code> interface was changed. </li>
+            <li> If you register custom metrics, you will need to update this code, because the <code>StreamsMetric</code> interface was changed. </li>
+            <li> Upgrading your Streams application from 0.10.1 to 0.10.2 does not require to upgrade your brokers.
+                 A Kafka Streams 0.10.2 application can connect to 0.10.2 and 0.10.1 brokers (it is not possible to connect to 0.10.0 brokers though). </li>
         </ul>
     </li>
-</ol>
+
+    <li>Streams API changes in 0.10.2.0:
+         <li> New methods in <code>KafkaStreams</code>:
+             <ul>
+                 <li> set a listener to react on application state change via <code>#setStateListener(StateListener listener)</code> </li>
+                 <li> retrieve the current application state via <code>#state()</code> </li>
+                 <li> retrieve the global metrics registry via <code>#metrics()</code> </li>
+                 <li> apply a timeout when closing an application via <code>#close(long timeout, TimeUnit timeUnit)</code> </li>
+                 <li> specify a custom indent when retrieving Kafka Streams information via <code>#toString(String indent)</code> </li>
+             </ul>
+         </li>
+         <li> Parameter updates in <code>StreamsConfig</code>:
+             <ul>
+                 <li> parameter "zookeeper.connect" got deprecated </li>
+                 <li> added many new parameters for metrics, security, and client configurations </li>
```

Comment:
This is a bit unclear: actually we only added configs (8 in total) for the admin client, and did not have new configs for metrics and security. It's better to enumerate them here and refer to the configs section for details.